express-notes-api/server.js

```diff
@@ -0,0 +1,184 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+/* This just makes it easier to create error messages for me.
+   And it looks cleaner when written in the code later on. */
+class Error {
+  constructor(errorMsg) {
+    this.error = errorMsg;
+  }
+}
+
+const port = 8080;
+const readFile = './data.json';
+const writeFile = './data.json';
+const app = express();
+const parser = express.json();
+app.use(parser);
+
+/* An async function for grabbing the data from the json file.
+   Data will be parsed before returning.                    */
+async function getJSON(fileUrl) {
+  try {
+    const contents = await fs.readFile(fileUrl, 'utf8');
+    const parsed = JSON.parse(contents);
+    return parsed;
+  } catch (err) {
+    console.log('Error reading json:', err);
+  }
```

Comment:
You do not want to catch the error in this function. Instead, you want to let it propagate. Here's why:

If `readFile` throws an exception, it will be caught here and logged, then the function will continue executing and return `undefined` since there is not a return statement after the catch block.

The caller will then receive `undefined` as the return value and will most likely crash.

So remove the try/catch block from this function and let the exception propagate up to the caller, which has a try/catch around it that will properly catch it and handle it.

The same applies to `writeJSON` below.